pkg/nar/dump.go

```diff
@@ -118,6 +143,8 @@ func dumpPath(nw *Writer, path string, subpath string) error {
 		}
 
 		return nil
+
+	case TypeUnknown:
```

Comment:
Move the `return fmt.Errorf("invalid mode for file %v", p)` from below?

Reply:
I tried that first, but then we'll still need to return the error at the end of the function to appease the compiler.
To avoid duplication I just let it fall through.

Reply:
`golangci-lint` requires all enum members to be present, so we're back to having to duplicate the error return in that case.

Reply:
golangci-lint requires the switch to be exhaustive (ie contain `TypeUnknown`), so we're back to duplicating the error return or opt out of the linter.
I would rather just stick the error at the end of the function.